plugins/module_utils/eda_module.py

```diff
@@ -759,7 +759,7 @@ def fail_wanted_one(self, response, endpoint, query_params):
     def get_exactly_one(self, endpoint, name_or_id=None, **kwargs):
         return self.get_one(endpoint, name_or_id=name_or_id, allow_none=False, **kwargs)
 
-    def resolve_name_to_id(self, endpoint, name_or_id, data):
+    def resolve_name_to_id(self, endpoint, name_or_id, data=None):
```

Comment:
I've made a minor change here as the sanity check prefers this to be `None` rather than `{}`